### ISSUE TITLE
Performance issue

### ISSUE THREAD

**Issue by achatina** [kizitonwose/CalendarView#181]
Hi there. 
Great lib, but it has really big problems with optimizations on long-range setup (e.g. -100 years, +100 years) because of `generateWeekDays`. It calls for each month and loads main thread. On a weak device it could take up to 5-10 seconds, which looks not very good. Seems like you need to review this approach. 
You can find it in Example3 if you set up it like this:
```
 setup(currentMonth.minusYears(100), currentMonth.plusYears(100), daysOfWeek.first())
```

![Снимок экрана от 2020-07-15 14-40-42](https://user-images.githubusercontent.com/16122829/87541773-7e10ed80-c6aa-11ea-8ec1-6379a121c9da.png)
![Снимок экрана от 2020-07-15 14-24-08](https://user-images.githubusercontent.com/16122829/87541782-82d5a180-c6aa-11ea-8937-7bb65c6b9845.png)
![Снимок экрана от 2020-07-15 14-49-11](https://user-images.githubusercontent.com/16122829/87541787-836e3800-c6aa-11ea-9e2a-f71aee7b45ec.png)

**Comment by kizitonwose** [comment]
You are right. I have been considering a `setupAsync()` alternative to the `setup()` method that basically does the date generation in the background and updates the calendar when ready. Just haven't got the time to actually implement it. Hopefully this weekend or the next one 🤞.

**Comment by achatina** [comment]
It would really help. Moreover, I want to suggest to you, check generating months dynamically after user's swipe. It could also become a part of infinite calendar. But it seems that it need to rewrite lot of logic, so mb think about it for far future

**Comment by bmilovanovic** [comment]
First of all, thank you @kizitonwose for such a wonderful library. I appreciate the time that you spent on making so many things customizable. Unfortunately, I must say that the performance is a dealbraker for me. I have a BottomNavigationView in my app, and when I switch to the CalendarView, it takes a whole lot for the tab to be displayed. I did not dive into the code so I don't know what is the problem here, but I may assume that you're generating all the months and days that are not visible (I concluded this because the scrolling is super smooth). Is there a way for me to configure this? The scrolling isn't important for me. Lazy loading is definitely what you should do here.

**Comment by kizitonwose** [comment]
> I want to suggest to you, check generating months dynamically after user's swipe. It could also become a part of infinite calendar. But it seems that it need to rewrite lot of logic, so mb think about it for far future

Yep, this is on my TODO, just need to find the time.

**Comment by kizitonwose** [comment]
>  Is there a way for me to configure this? The scrolling isn't important for me. Lazy loading is definitely what you should do here.

Not at the moment but the async setup should come this weekend(hopefully), and that should be sufficient until lazy loading is possible.

**Comment by michaeltys** [comment]
Would be really great to get this feature

**Comment by kizitonwose** [comment]
Happy to announce that `setupAsync` is now available in [0.4.1](https://github.com/kizitonwose/CalendarView/releases/tag/0.4.1). 
There are also other async methods for related tasks like updating the start and end months, etc.
